SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -50,87 +52,110 @@ bool FSpatialGDKEditor::GenerateSchema(bool bFullScan)
 		}
 	}
 
-	bSchemaGeneratorRunning = true;
-
-	// 80/10/10 load assets / gen schema / garbage collection.
-	FScopedSlowTask Progress(100.f, LOCTEXT("GeneratingSchema", "Generating Schema..."));
-	Progress.MakeDialog(true);
-
-	RemoveEditorAssetLoadedCallback();
-
 	if (Schema::IsAssetReadOnly(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
 	{
-		bSchemaGeneratorRunning = false;
 		return false;
 	}
 
-	if (!Schema::LoadGeneratorStateFromSchemaDatabase(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
+	if (Method == FullAssetScan)
 	{
-		Schema::ResetSchemaGeneratorStateAndCleanupFolders();
-	}
+		// Make sure SchemaDatabase is not loaded.
+		if (UPackage* LoadedDatabase = FindPackage(nullptr, *FPaths::Combine(TEXT("/Game/"), *SpatialConstants::SCHEMA_DATABASE_FILE_PATH)))
+		{
+			TArray<UPackage*> ToUnload;
+			ToUnload.Add(LoadedDatabase);
+			UPackageTools::UnloadPackages(ToUnload);
+		}
 
-	TArray<TStrongObjectPtr<UObject>> LoadedAssets;
-	if (bFullScan)
-	{
-		Progress.EnterProgressFrame(80.f);
-		if (!LoadPotentialAssets(LoadedAssets))
+		const USpatialGDKEditorSettings* EditorSettings = GetDefault<USpatialGDKEditorSettings>();
+
+		const FString& PlatformName = EditorSettings->GetCookAndGenerateSchemaTargetPlatform();
+
+		if (PlatformName.IsEmpty())
```

Comment:
How about we just assume LinuxServer here, I've never seen someone use anything else

Reply:
Well, I do. I often build the engine forgetting to setup the linux tool path. So at least the default command will work on everyone's setup.